src/main/scala/com/codacy/plugins/results/traits/ToolDocumentation.scala

```diff
@@ -0,0 +1,39 @@
+package com.codacy.plugins.results.traits
+
+import com.codacy.plugins.api.PatternDescription
+import com.codacy.plugins.api.results.Tool
+import com.codacy.plugins.results.docker.utils.PluginPrefixHelper
+import com.codacy.plugins.runners.IDocker
+import com.codacy.plugins.utils.DockerHelper
+
+trait ToolDocumentation {
+  def toolSpecification: Option[Tool.Specification]
+  def patternDescriptions: Option[Set[PatternDescription]]
+  def toolPrefix: String
+  def toolDescription: Option[String]
+
+  // Prefixed versions of the private instances above
+  lazy val prefixedSpecs: Option[Tool.Specification] = PluginPrefixHelper.prefixSpec(toolPrefix, toolSpecification)
+  lazy val prefixedDescriptions: Option[Set[PatternDescription]] =
+    PluginPrefixHelper.prefixDescription(toolPrefix, patternDescriptions)
+}
+
+@deprecated(
```

Comment:
Is this a copy paste form the codacy-plugins or something? Or what is the idea of this deprecated?